pkg/controller/machinepool/machinepool_controller_test.go

```diff
@@ -497,6 +497,80 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
+				testMachineAutoscaler("foo-12345-worker-us-east-1b", "1", 1, 2),
+				testMachineAutoscaler("foo-12345-worker-us-east-1c", "1", 1, 1),
+			},
+			generatedMachineSets: []*machineapi.MachineSet{
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", false, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", false, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1c", "worker", false, 1, 0),
+			},
+			expectedRemoteMachineSets: []*machineapi.MachineSet{
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0),
+			},
+			expectedRemoteMachineAutoscalers: []autoscalingv1beta1.MachineAutoscaler{
+				*testMachineAutoscaler("foo-12345-worker-us-east-1a", "1", 1, 2),
+				*testMachineAutoscaler("foo-12345-worker-us-east-1b", "1", 1, 2),
+				*testMachineAutoscaler("foo-12345-worker-us-east-1c", "1", 1, 1),
+			},
+			expectedRemoteClusterAutoscalers: []autoscalingv1.ClusterAutoscaler{
+				*testClusterAutoscaler("2"),
+			},
+		},
+		{
+			name:              "Update cluster autoscaler when BalanceSimilarNodeGroups disabled",
+			clusterDeployment: testClusterDeployment(),
+			machinePool:       testAutoscalingMachinePool(3, 5),
+			remoteExisting: []runtime.Object{
+				testMachine("master1", "master"),
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0),
+				func() runtime.Object {
+					a := testClusterAutoscaler("1")
+					a.Spec.BalanceSimilarNodeGroups = pointer.BoolPtr(false)
+					return a
+				}(),
+				testMachineAutoscaler("foo-12345-worker-us-east-1a", "1", 1, 2),
+				testMachineAutoscaler("foo-12345-worker-us-east-1b", "1", 1, 2),
+				testMachineAutoscaler("foo-12345-worker-us-east-1c", "1", 1, 1),
+			},
+			generatedMachineSets: []*machineapi.MachineSet{
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", false, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", false, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1c", "worker", false, 1, 0),
+			},
+			expectedRemoteMachineSets: []*machineapi.MachineSet{
+				testMachineSet("foo-12345-worker-us-east-1a", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1b", "worker", true, 1, 0),
+				testMachineSet("foo-12345-worker-us-east-1c", "worker", true, 1, 0),
+			},
+			expectedRemoteMachineAutoscalers: []autoscalingv1beta1.MachineAutoscaler{
+				*testMachineAutoscaler("foo-12345-worker-us-east-1a", "1", 1, 2),
+				*testMachineAutoscaler("foo-12345-worker-us-east-1b", "1", 1, 2),
+				*testMachineAutoscaler("foo-12345-worker-us-east-1c", "1", 1, 1),
+			},
+			expectedRemoteClusterAutoscalers: []autoscalingv1.ClusterAutoscaler{
+				*testClusterAutoscaler("2"),
+			},
+		},
```

Comment:
Do we need to add logic to the test driver to assert that the field is set? I don't see anywhere we're doing that at the moment.

Reply:
The autoscaler returned from `testClusterAutoscaler()` has `BalanceSimilarNodeGroups = true` so when we test the expected clusterautoscaler below we are ensuring it is set.

Reply:
Cool. The additional test case (starting/staying disabled) makes this clearer as well. Thanks.